src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -20,148 +16,201 @@
-	private static final double steerPositionCoefficient =
-			(ticksPerRotation / (2 * Math.PI)) * steerReduction; // radians
-	// per
-	// tick
-	private static final double driveVelocityCoefficient =
-			(ticksPerRotation / (Math.PI * wheelDiameterMeters))
-					* driveReductionL1; // ticks per meter per 100 ms
-
-	private static final int talonFXLoopNumber = 0;
-	private static final int canTimeoutMS = 20;
-
-	private static PFFController<Double> balanceController;
-
-	private static final double tipF = 0.01;
-	private static final double tipP = 0.05;
-	private static final double tipTolerance = 5;
-
-	private WPI_TalonFX[] moduleDriveMotors = {
-		new WPI_TalonFX(Hardware.DRIVEBASE_FRONT_LEFT_DRIVE_MOTOR),
-		new WPI_TalonFX(Hardware.DRIVEBASE_FRONT_RIGHT_DRIVE_MOTOR),
-		new WPI_TalonFX(Hardware.DRIVEBASE_BACK_LEFT_DRIVE_MOTOR),
-		new WPI_TalonFX(Hardware.DRIVEBASE_BACK_RIGHT_DRIVE_MOTOR)
-	};
+	private static final boolean IS_COMP = Robot.getInstance().isCompetition();
 
-	private WPI_TalonFX[] moduleAngleMotors = {
-		new WPI_TalonFX(Hardware.DRIVEBASE_FRONT_LEFT_ANGLE_MOTOR),
-		new WPI_TalonFX(Hardware.DRIVEBASE_FRONT_RIGHT_ANGLE_MOTOR),
-		new WPI_TalonFX(Hardware.DRIVEBASE_BACK_LEFT_ANGLE_MOTOR),
-		new WPI_TalonFX(Hardware.DRIVEBASE_BACK_RIGHT_ANGLE_MOTOR)
+	// ordered from front left, front right, back left, back right
+	private static final Rotation2d[] PRACTICE_DRIVEBASE_ENCODER_OFFSETS = {
+		Rotation2d.fromDegrees(337.236),
+		Rotation2d.fromDegrees(251.982),
+		Rotation2d.fromDegrees(205.839),
+		Rotation2d.fromDegrees(311.396)
+	};
+	private static final Rotation2d[] COMP_DRIVEBASE_ENCODER_OFFSETS = {
+		Rotation2d.fromDegrees(-111.796),
+		Rotation2d.fromDegrees(-343.388),
+		Rotation2d.fromDegrees(-21.796),
+		Rotation2d.fromDegrees(-332.841)
 	};
 
-	private WPI_CANCoder[] moduleEncoders = {
+	// max drive speed is from SDS website and not calculated with robot weight
+	public static final double MAX_DRIVE_SPEED_METERS_PER_SEC = IS_COMP ? 4.4196 : 4.1148;
+	// see how this is calculated here: https://www.desmos.com/calculator/rrglfgzqnh
+	// it's basically just rotations_per_sec = velocity/(2*pi*turning radius)
+	public static final Rotation2d MAX_ROTATIONS_PER_SEC =
+			Rotation2d.fromRotations(IS_COMP ? 1.171 : 2.303);
+
+	// magic number found by trial and error (aka informal characterization)
+	private static final double ODOMETRY_ADJUSTMENT = 0.957;
+	// 4 inches * odemetry adjustment
+	private static final double WHEEL_DIAMETER_METERS = 0.1016 * ODOMETRY_ADJUSTMENT;
+	private static final double DRIVE_REDUCTION = IS_COMP ? 6.75 : 8.14;
+	// steer reduction is the conversion from rotations of motor to rotations of the wheel
+	// module rotation * STEER_REDUCTION = motor rotation
+	public static final double STEER_REDUCTION =
+			IS_COMP ? 150.0 / 7.0 : (32.0 / 15.0) * (60.0 / 10.0);
+
+	private static final double TIP_F = 0.01;
+	private static final double TIP_P = 0.05;
+	private static final double TIP_TOLERANCE = 5;
```

Comment:
Assuming this is degrees, this seems pretty high if the tolerance of the official charge station is 2.5 degrees?
Or is this because our charge station is not level?
We should note this needs to be updated or verified before the first competition.

Reply:
It is in degrees, and yes updating before comp is a good idea, especially once we have the comp robot fully constructed and can test with its full weight

Reply:
Issue added #35